chain/default_syncer.go

```diff
@@ -133,7 +163,11 @@ func (syncer *DefaultSyncer) collectChain(ctx context.Context, tipsetCids types.
 	fetchedHead := tipsetCids
 	defer logSyncer.Infof("chain fetch from network complete %v", fetchedHead)
 
-	for {
+	// Continue collecting the chain if we're either not yet caught up or the
+	// height of the input blocks has not yet exceeded the sum of the current
+	// consensus height and the finalityLimit constant, otherwise ignore the input
+	// blocks as a likely invalid chain or denial of service attempt.
+	for syncer.SyncMode == Syncing || len(chain) < FinalityLimit {
```

Comment:
Please note that this is number of blocks and not block height people.  I think this is the right thing to do right now, but this will trip us up if we don't internalize it.

Reply:
Blocking: well spotted. The issues for this work identify number of rounds, not blocks. Please use block height arithmetic instead of `len()`.